generic/tests/nfs_corrupt.py

```diff
@@ -194,14 +194,17 @@ def check_vm_status(vm, status):
         config.setup()
     except NFSCorruptError as e:
         test.error(str(e))
-    image_name = os.path.join(config.mnt_dir, "nfs_corrupt")
+
+    image_name = "nfs_corrupt"
     params["image_name_stg"] = image_name
-    params["force_create_image_stg"] = "yes"
-    params["create_image_stg"] = "yes"
+    image_stg_dir = config.mnt_dir
     stg_params = params.object_params("stg")
+    stg_img = QemuImg(stg_params, image_stg_dir, image_name)
+    stg_img.create(stg_params)
 
     error_context.context("Boot vm with image on NFS server", logging.info)
-    env_process.preprocess_image(test, stg_params, image_name)
+    image_name = os.path.join(config.mnt_dir, "nfs_corrupt")
+    params["image_name_stg"] = image_name
```

Comment:
just FYI. 

It's hard to describe this, let me have a try:

The way you did:
1. stg image might not be removed after testing, for the nfs mnt has already been un-mounted before post_process_image (please double check your log)
2. it's a little difficult to understand your progress, I think you'd like to let QemuImg and VT handle the things in a correct way, so you made several assignments for params['image_name_stg']

Another way: just keep the original process steps, there is a param 'images_base_dir' to define the root _dir for image, so we can add params['images_base_dir_stg'] = config.mnt_dir, then preprocess_image, the stg image can be created by vt successfully.

also we still need to create the stg_img object in order to check_image, and what's more, we have to use stg_img.remove() to remove stg image instead of vt's image removal, that's because there might be a bug in vt, I just found, when post_process_image in vt, vt get root_dir as data_dir.get_data_dir directly(data_dir.get_data_dir()) , instead of getting 'images_base_dir' first (params.get("images_base_dir", data_dir.get_data_dir())), so stg never removed.

Both ways are OK for me, but in your way, it's better to remove the image by stg_img.remove(), and at line 202, we use tag to create QemuImg object instead of image_name, e.g.  stg_img = QemuImg(stg_params, image_stg_dir, 'stg')

Reply:
> just FYI.
> 
> It's hard to describe this, let me have a try:
> 
> The way you did:
> 
> 1. stg image might not be removed after testing, for the nfs mnt has already been un-mounted before post_process_image (please double check your log)

Yes, you are right, I added stg_img.remove()

> 2. it's a little difficult to understand your progress, I think you'd like to let QemuImg and VT handle the things in a correct way, so you made several assignments for params['image_name_stg']

It's a workaround of luks bug in vt, It's not worth fixing in vt.

> 
> Another way: just keep the original process steps, there is a param 'images_base_dir' to define the root _dir for image, so we can add params['images_base_dir_stg'] = config.mnt_dir, then preprocess_image, the stg image can be created by vt successfully.
> 
> also we still need to create the stg_img object in order to check_image, and what's more, we have to use stg_img.remove() to remove stg image instead of vt's image removal, that's because there might be a bug in vt, I just found, when post_process_image in vt, vt get root_dir as data_dir.get_data_dir directly(data_dir.get_data_dir()) , instead of getting 'images_base_dir' first (params.get("images_base_dir", data_dir.get_data_dir())), so stg never removed.
> 

Yes, you are right, using images_base_dir_stg  the stg image can be created by vt successfully. But it doesn't return value in preprocess_image and stg can't remove by post_process_image. So I will keep my way, many thanks. 

> Both ways are OK for me, but in your way, it's better to remove the image by stg_img.remove(), and at line 202, we use tag to create QemuImg object instead of image_name, e.g. stg_img = QemuImg(stg_params, image_stg_dir, 'stg')

Updated, many thanks.